### PR TITLE
Marshalling Issues - Multiple Issues related to problems with marshallers.

### PR DESCRIPTION
**Pull request by Josephblt**
[KOGITO-7531 - SWF Editor - End construct support](https://issues.redhat.com/browse/KOGITO-7531)
[KOGITO-7591 - SWF Editor - Is adding extra "end": false to eventConditions fields even when transition is set](https://issues.redhat.com/browse/KOGITO-7591)
[KOGITO-7608 - SWF Editor - Is removing "schedule" from "start" object and makes is a string](https://issues.redhat.com/browse/KOGITO-7608)
[KOGITO-7609 - SWF Editor - Is changing transition object to string](https://issues.redhat.com/browse/KOGITO-7609)
[KOGITO-7599 - SWF Editor - Cannot open workflows from specification examples](https://issues.redhat.com/browse/KOGITO-7599)
[KOGITO-7594 - SWF Editor - Is unable to load SendCloudEventOnProvisionExample.sw.json](https://issues.redhat.com/browse/KOGITO-7594)

VS Code Extension: [vscode_extension_serverless_workflow_editor_0.0.0.vsix](https://drive.google.com/file/d/1T2HhNPg_I6n6uVxBtVChyoO0xeOHOIkM/view?usp=sharing)

### CODE REVIEW

**Comment by domhanak** [comment]
@Josephblt Could you please enable the IT test suite for `serverless-workflow-diagram-editor`? In `package.json`
Replace `-DskipITs ` part with `-DskipITs=$(build-env global.build.testIT --not)`

So that we validate the issues.

**Comment by domhanak** [comment]
@Josephblt So, looking at the logs, we have 3 errors and one failure. For errors, we can;t still load the files, is that expected? I remember you mentioned this does not fix all, so maybe just ignore those test cases and lets proceed. Anywa the number of fails/errors is lower which is awesome! :) 
```
2022-08-22T15:52:42.9209445Z [ERROR] Failures: 
2022-08-22T15:52:42.9210553Z [ERROR]   SWEditorSeleniumIT.testEventBasedServiceInvocation:206->testExample:256 JSON documents are different:
2022-08-22T15:52:42.9211958Z Different value found in node "states[0].actions[0]", expected: <{"name":"MakeAppointmentAction","eventRef":{"produceEventRef":"MakeVetAppointment","data":"${ .patientInfo }","consumeEventRef":"VetAppointmentInfo"},"actionDataFilter":{"results":"${ .appointmentInfo }"}}> but was: <null>.
2022-08-22T15:52:42.9212532Z 
2022-08-22T15:52:42.9212662Z [ERROR] Errors: 
2022-08-22T15:52:42.9226740Z [ERROR]   SWEditorSeleniumIT.testBookLendingExample:231->testExample:253->waitCanvasPanel:278 » Timeout
2022-08-22T15:52:42.9242604Z [ERROR]   SWEditorSeleniumIT.testCustomerCreditCheckExample:166->testExample:253->waitCanvasPanel:278 » Timeout
2022-08-22T15:52:42.9243674Z [ERROR]   SWEditorSeleniumIT.testJobMonitoringExample:171->testExample:253->waitCanvasPanel:278 » Timeout
```

**Comment by Josephblt** [comment]
Hello @domhanak. I'm investigating the failure. The testEventBasedServiceInvocation has differences so I plan on fixing that.
But the other ones are strange. I tested them and they open just fine after Kirrill's fix, so I have to further investigate. Another thing I did was to send a PR to fix the BookLendingExample. 

Here it is:
https://github.com/serverlessworkflow/specification/pull/663.

I'll update this PR when I have a fix at least for the failing example. Thanks.

**Comment by domhanak** [comment]
@Josephblt Ack, let me know when ready

**Comment by Josephblt** [comment]
Hello @domhanak. Finally... I managed to fix all issues. Even the action support part. Although we still do not support Actions entirely, we can serialize/deserialize it properly. Other issues were fixed and we can run all examples without problems. If you can take a look at this so we can get it merged, that would be great. Thanks.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java [hunk 1]
```diff
@@ -81,9 +84,16 @@ public static void setupClass() {
...
     public void openSWEditor() {
+        //For Firefox Testing
         final FirefoxOptions firefoxOptions = new FirefoxOptions();
         firefoxOptions.setHeadless(HEADLESS);
         driver = new FirefoxDriver(firefoxOptions);
+
+        //For Chrome Testing
+//        final ChromeOptions chromeOptions = new ChromeOptions();
+//        chromeOptions.setHeadless(true);
+//        driver = new ChromeDriver(chromeOptions);
```
Comment (domhanak) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java]: I guess we can remove this for now, subsequent work on IT tests will lead to creating something like SWEditorBaseIT which will differentiate Firefox/Chrome.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java [hunk 2]
```diff
@@ -284,7 +294,10 @@ private void waitCanvasPanel() {
 
     private void setContent(final String xml) {
         try {
-            ((JavascriptExecutor) driver).executeScript(String.format(SET_CONTENT_TEMPLATE, xml));
+            String content = String.format(SET_CONTENT_TEMPLATE, xml);
+            content = content.replace(CONTENT_REGULAR_CHAR, CONTENT_EXECUTOR_CHAR);
+            ((JavascriptExecutor) driver).executeScript(content);
```
Comment (domhanak) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java:292]: Why is this necessary?  Like if I need to escape some characters in my content before it can be set, that does not sound right. Shouldn't the escaping/replacing be done in the `setContent` context? e.g. setContent validates/scapes inputs or something. I should not be needed to do so.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java (hunk 1 above)
Comment (Josephblt) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java]: Sorry I intended to remove this and forgot. Firefox driver did not work on my computer. Had to use Chrome Driver, which worked well even directly on IntelliJ.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java (hunk 2 above)
Comment (Josephblt) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWEditorSeleniumIT.java:292]: This is necessary due to escaping characters within the string. Take the BookLendingExample for instance,
this [data condition](https://github.com/domhanak/kie-tools/blob/654af8a2dd1830f480114631bc1e89b34fab5558/packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/resources/org/kie/workbench/common/stunner/sw/client/selenium/BookLendingExample.sw.json#L39) has escaped characters (double quotations). This causes problems when directly calling setContent with that string. So the conversion is necessary, otherwise, the parser is not able to properly nest the double-quotations.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-api/src/main/java/org/kie/workbench/common/stunner/sw/marshall/TransitionMarshalling.java [hunk 3]
```diff
@@ -59,19 +61,26 @@ public interface TransitionMarshalling {
...
                             sourceState.transition = null;
-
                             if (sourceState.end instanceof Boolean) {
                                 sourceState.end = true;
-                            } // else is Object
+                            }
                         } else {
-                            sourceState.transition = getStateNodeName(targetNode);
-                            sourceState.end = false;
+                            if (sourceState.transition instanceof String) {
```
Comment (domhanak) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-api/src/main/java/org/kie/workbench/common/stunner/sw/marshall/TransitionMarshalling.java:71]: Is there any unit test associated with this class & method? I see no updates related to this in unit tests, so it suggest to me we do not have unit test for this class.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-api/src/main/java/org/kie/workbench/common/stunner/sw/marshall/WorkflowMarshalling.java [hunk 4]
```diff
@@ -72,7 +72,7 @@ public interface WorkflowMarshalling {
                     Edge startEdge = (Edge) startNode.getOutEdges().get(0);
                     marshallEdge(context, startEdge);
                 } else {
-                    workflow.setStart("");
+                    workflow.setStart(null);
```
Comment (domhanak) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-api/src/main/java/org/kie/workbench/common/stunner/sw/marshall/WorkflowMarshalling.java:75]: Is there any unit test associated with this class & method? I see no updates related to this in unit tests, so it suggest to me we do not have unit test for this class.

**Comment by romartin** [comment]
@romartin missing to do the review

**Comment by romartin** [comment]
Hey @Josephblt @domhanak 

I can't find the artifacts for manual testing being generated in this PR. You know guys where I can download them? Have the location been changed, as far as I remember they were available in the "checks" tab, but actuallly nothing there...

thx in advance

@tiagobento ^^

**Comment by romartin** [comment]
BTW @Josephblt @domhanak this PR is a blocker for the DP2 release. It should be merged during this week, as far as possible. So can you please take it as a priority?? 

Also please do not just focus on testing the PR itself, would be nice if you can also perform some general smoke testing. This is due to this editor will became the default one on the next release. so exposed to the word, and I would like to have it properly (manual) tested as well.

@Josephblt can you please upload the vsix for testing, in case no being auto-generated? I will help on the smoke testing guys once having the vsix.

Thx in adance!

**Comment by romartin** [comment]
@Josephblt Sorry I just see the vsix is being uploaded. Anyway, any ideas why it's not being auto-generated ?

**Comment by Josephblt** [comment]
Hello @romartin, @domhanak. No idea why the .vsix is not auto-generated. But anyway, I always put the extension up top in the first comment. I also update it today, so we have it rebased with the latest changes.

**Review by romartin — APPROVED** [review]
Thanks @Josephblt .

I have tested the app and everything works properly. Approving! IMO we can merge this, and go for DP2 :) 

Although, can you please sync with @domhanak in order to see if now all examples from our IT tests are working? Also, please as commented, can you please open a new PR for testing properly all these changes in our integration tests as well?

Also please @domhanak would appreciate your feedback here during this week.

Thx guys!

**Comment by Josephblt** [comment]
Hello @romartin. All example tests are working. The automated builds are already checking for them.

**Comment by domhanak** [comment]
Ticket for missing unit tests - https://issues.redhat.com/browse/KOGITO-7902

**Review by domhanak — APPROVED** [review]
Approving, the issues are resolved, integration tests are running and pass.

The fact we do not have a unit tests is bad, reported in https://issues.redhat.com/browse/KOGITO-7902 and we should fix this ASAP. I do not want to block this PR because of this as it is a valuable contribution and not a cause the test are missing.

**Comment by romartin** [comment]
I am absolutely agree @domhanak . Those unit tests for the marshalling are really important, and also because we'll move to new parsers in short term.

Please @Josephblt can you focus on creating unit tests for all those marshalling updates?

Thanks guys!
